### PR TITLE
feat(ion-identity-client): Add proper private key encryption/decryption

### PR DESCRIPTION
**Pull request by ice-tychon**
### What does this PR do?
This PR adds a proper private key encryption/decryption. It's located in the `KeyService`.
Also, I've refactored some existing code to match FP patterns (now failures are properly processed)

#### Minor changes related to DataSources
- Logic inside of `CreateRecoveryCredentialsDataSource ` is not changed, only return types from `NetworkFailure` to `CreateRecoveryCredentialsFailure `.
- The same thing for `RecoverUserDataSource` – changed return types.
- Also `CreateRecoveryCredentialsFailure` and `RecoverUserDataSource` now have more specific failures.

#### Changes related to Services
- Extract key-related logic from `CreateRecoveryCredentialsService` and `RecoverUserService` to `KeyService`
- Changed logic of `generateRecoveryCode`, so the recovery code is a real thing now (e.g. `@{z,96_]U6AM>oEPlJjAF.?OC,S=v~W1`)
- `generateCredentialName` now uses UUID v4 (e.g. `E86B531D-5B9C-40B8-A64C-8AFAE5391AAE`)
- `generateCredId` now returns a more beautiful string (e.g. `AT5PR-JGD4J-37NJ0-WW4QF-6CLP5`)

#### Other changes
Other changes are related to generated files. You can skip them.

### CODE REVIEW

File: packages/ion_identity_client/lib/src/auth/services/create_recovery_credentials_service.dart [hunk 1]
```diff
@@ -209,62 +206,42 @@ class CreateRecoveryCredentialsService {
...
-    final privateKeyInfoSeq = ASN1Sequence()
-      ..add(ASN1Integer(BigInt.zero)) // Version
-      ..add(algorithmSeq)
-      ..add(ASN1OctetString(privateKeyOctetString.encodedBytes));
-
-    final bytes = privateKeyInfoSeq.encodedBytes;
-    final base64Str = base64.encode(bytes);
-
-    return '-----BEGIN PRIVATE KEY-----\n${chunk(base64Str)}\n-----END PRIVATE KEY-----';
+  String generateRecoveryCode() {
```
[the remark this answers is not in the document]
Comment (ice-tychon) [reply packages/ion_identity_client/lib/src/auth/services/create_recovery_credentials_service.dart:226]: Done. Example of a new recovery code: 81c63bab-19fd6c57-a14d3d4e-f3706e0d

File: packages/ion_identity_client/lib/src/auth/services/create_recovery_credentials_service.dart [hunk 2]
```diff
@@ -1,68 +1,86 @@
-// SPDX-License-Identifier: ice License 1.0
```
Comment (ice-alcides) [inline packages/ion_identity_client/lib/src/auth/services/create_recovery_credentials_service.dart:1]: Need to add a License header to all files

File: packages/ion_identity_client/lib/src/auth/services/create_recovery_credentials_service.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (ice-tychon) [reply packages/ion_identity_client/lib/src/auth/services/create_recovery_credentials_service.dart:227]: Updated once more. Example of a password (recovery code): `@{z,96_]U6AM>oEPlJjAF.?OC,S=v~W1`

File: packages/ion_identity_client/lib/src/auth/services/create_recovery_credentials_service.dart (hunk 2 above)
Comment (ice-tychon) [reply packages/ion_identity_client/lib/src/auth/services/create_recovery_credentials_service.dart:1]: Added license to all files without a license (skipped `.g.dart` files)

**Review by ice-alcides — APPROVED** [review]

**Review by ice-alcides — APPROVED** [review]

**Pull request merged**